lib/galaxy/webapps/galaxy/controllers/admin_toolshed.py

```diff
@@ -2059,11 +2059,12 @@ def view_tool_metadata( self, trans, repository_id, tool_id, **kwd ):
                     tool_config = tool_metadata[ 'tool_config' ]
                     if shed_config_dict and shed_config_dict.get( 'tool_path' ):
                         tool_config = os.path.join( shed_config_dict.get( 'tool_path' ), tool_config )
-                    tool = trans.app.toolbox.load_tool( os.path.abspath( tool_config ), guid=tool_metadata[ 'guid' ] )
-                    if tool:
-                        tvm = tool_version_manager.ToolVersionManager( trans.app )
-                        tool_version = tvm.get_tool_version( str( tool.id ) )
-                        tool_lineage = tool_version.get_version_ids( trans.app, reverse=True )
+                    tool = trans.app.toolbox.get_tool(tool_id=tool_metadata[ 'guid' ], exact=True)
+                    if not tool:
+                        tool = trans.app.toolbox.load_tool( os.path.abspath( tool_config ), guid=tool_metadata[ 'guid' ] )
+                        if tool:
+                            tool._lineage = trans.app.toolbox._lineage_map.register( tool )
+                    tool_lineage = tool.lineage.get_version_ids(reverse=True)
```

Comment:
Then we wouldn't get the lineage if the tool was already in the toolbox.
I have added another `if tool:` statement in 1e11c46c348ad65835bda2cb5468c7b3e3b2e303